### DISCUSSION TITLE
Shipment APIs

### DISCUSSION THREAD

**Discussion by FloppyChips** [amzn/selling-partner-api-models#3557] [General]
I have an SP-API developer account, have developed an app and this is active for listings and orders.

We want to add shipping, initially the direct-to-consumer role was denied, ok fine because of PII.

But how do we integrate with Shipping v2 API providing the from address instead when either this role or the Amazon Logistics role are required and I am told by support that Amazon Logistics is invite only?  Is there a way or should I be looking at a 3rd party integration instead. I don't know why Amazon would turn away business in this way.

Thanks Stuart

**Comment by MikeWillis** [comment]
did you ever find a solution to this? We are not even seeing Amazon Logistics as an option in the roles list

**Reply by FloppyChips** [reply]
No, this is what Amazon support said:

"This is Rick, from the Selling Partner API Developer Support (SPADS) team.

Our Internal team has provided documentation as to why you are unable to select the role "Amazon Logistics".

This role is only available for Developers that use the platform Developer Central (https://developercentral.amazon.com/ ) to create and edit their Apps and that are part of an invite-only program for using Amazon Shipping Services in 3p websites.

So this feature is not currently available for the public.

We recommend staying in the documentation for Selling Partner API, as the Business APIs and Shipping APIs Documentation has many Invite only features that are not available for regular Selling Partner API developers.

We apologize for the inconvenience."

**Reply by MikeWillis** [reply]
ugh, ok thanks... It's very confusing. We're trying to purchase shipping labels for Amazon orders, not our own 3p website.

**Reply by chengpan9** [reply]
Can you check the Amazon Logistics role now?

**Comment by sungolivia** [comment]
hey yall it's me again! If you are interested in Amazon Shipping API v2, please send an email to amazonshippingintegrations@amazon.com and they should be able to help you out with it!

**Comment by MeenaAmz** [comment]
@chengpan9 This is an old discussion. We do have [shipping api](https://developer-docs.amazon.com/amazon-shipping/docs/shipping-api-v2-reference) available for everyone. To use the shipping API, you would need Product Listing or Direct to Consumer Shipping role ([Reference](https://developer-docs.amazon.com/sp-api/docs/role-mappings#shipping-api-v2)). Let me know if you have further questions.

Thanks
Meena
Selling Partner Developer Services
